src/libraries/System.Private.CoreLib/src/System/Environment.UnixOrBrowser.cs

```diff
@@ -0,0 +1,111 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.IO;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Text;
+using System.Threading;
+
+namespace System
+{
+    public static partial class Environment
+    {
+        public static bool UserInteractive => true;
+
+        private static string CurrentDirectoryCore
+        {
+            get => Interop.Sys.GetCwd();
+            set => Interop.CheckIo(Interop.Sys.ChDir(value), value, isDirectory: true);
+        }
+
+        private static string ExpandEnvironmentVariablesCore(string name)
+        {
+            var result = new ValueStringBuilder(stackalloc char[128]);
+
+            int lastPos = 0, pos;
+            while (lastPos < name.Length && (pos = name.IndexOf('%', lastPos + 1)) >= 0)
+            {
+                if (name[lastPos] == '%')
+                {
+                    string key = name.Substring(lastPos + 1, pos - lastPos - 1);
+                    string? value = GetEnvironmentVariable(key);
+                    if (value != null)
+                    {
+                        result.Append(value);
+                        lastPos = pos + 1;
+                        continue;
+                    }
+                }
+                result.Append(name.AsSpan(lastPos, pos - lastPos));
+                lastPos = pos;
+            }
+            result.Append(name.AsSpan(lastPos));
+
+            return result.ToString();
+        }
+
+        private static bool Is64BitOperatingSystemWhen32BitProcess => false;
+
+        private static int GetCurrentProcessId() => Interop.Sys.GetPid();
+
+        internal const string NewLineConst = "\n";
+
+        public static string SystemDirectory => GetFolderPathCore(SpecialFolder.System, SpecialFolderOption.None);
+
+        public static int SystemPageSize => CheckedSysConf(Interop.Sys.SysConfName._SC_PAGESIZE);
+
+        private static unsafe bool TryGetUserNameFromPasswd(byte* buf, int bufLen, out string? username)
```

Comment:
Why is it in UnixOrBrowser? It seems to be needed on for Unix

Reply:
Unless we're going to hard code it, it should be here.  Thoughts?

https://github.com/emscripten-core/emscripten/blob/master/system/lib/libc/musl/src/legacy/getpagesize.c#L5-L10

Reply:
This is about `string UserName`

Reply:
@steveisok this should still be moved to `Environment.Unix.cs`.